### PR TITLE
Enable asynchronous commit per connection level

### PR DESCRIPTION
**Pull request by tudor-da**
Currently, asynchronous commits are enabled at transaction level for the PostgreSQL JdbcLedgerDao. As a further optimization, this setting is moved at JDBC connection level (when Hikari spawns it).

This optimization aims at removing ~0.27 ms out of each transaction insertion for the indexer. At current performance levels, throughput improves by ~2 %.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [x] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/store/DbType.scala [hunk 1]
```diff
@@ -6,16 +6,17 @@ package com.daml.platform.store
 private[platform] sealed abstract class DbType(
     val name: String,
     val driver: String,
-    val supportsParallelWrites: Boolean)
+    val supportsParallelWrites: Boolean,
+    val supportsAsynchronousCommits: Boolean)
 
 private[platform] object DbType {
-  object Postgres extends DbType("postgres", "org.postgresql.Driver", true)
+  object Postgres extends DbType("postgres", "org.postgresql.Driver", true, true)
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/main/scala/platform/store/DbType.scala:13]: Can you use named parameters for the booleans here and below?

**Comment by SamirTalwar-DA** [comment]
The code looks good.

Do you have any way of testing whether the commits were _actually_ asynchronous, or do we have to test that manually? If so, how did you test it?

**Comment by tudor-da** [comment]
> The code looks good.
> 
> Do you have any way of testing whether the commits were _actually_ asynchronous, or do we have to test that manually? If so, how did you test it?

I've indirectly tested it by observing the throughput increase when comparing against the baseline. I didn't find a way to directly assert the nature of the commit mode in transactions. An indirect test would be to test the `HikariConnection` and see if the setting is off (e.g. `SHOW synchronous_commit`) for a spawned connection.

**Comment by SamirTalwar-DA** [comment]
That isn't a bad idea, though I can see it would be quite involved.

Can you at least see a difference in the PostgreSQL logs during a manual test?

**Review by SamirTalwar-DA — APPROVED** [review]

**Comment by tudor-da** [comment]
> That isn't a bad idea, though I can see it would be quite involved.
> 
> Can you at least see a difference in the PostgreSQL logs during a manual test?

Added a test for `HikariConnection` asserting the `synchronous_commit` mode.

**Review by mziolekda — APPROVED** [review]

**Pull request merged**